harness/determined/tensorboard/fetchers/base.py

```diff
@@ -1,28 +1,64 @@
 import abc
-from typing import Any, Callable, Dict, List
+import datetime
+from typing import Any, Callable, Dict, Generator, List
 
 
 class Fetcher(metaclass=abc.ABCMeta):
-    """Abstract base class for Tensorflow fetchers.
+    """Abstract base class for TensorBoard fetchers.
 
-    Responsible for syncing tensorflow files from a list of storage paths to a local directory.
+    Syncs TensorBoard files from remote file blob stores.
     """
 
     storage_paths: List[str]
+    _file_records: Dict[str, datetime.datetime] = {}
 
     @abc.abstractmethod
     def __init__(self, storage_config: Dict[str, Any], storage_paths: List[str], local_dir: str):
         pass
 
     @abc.abstractmethod
-    def fetch_new(self, new_file_callback: Callable = lambda: None) -> int:
-        """Fetches changed files found in storage paths to local disk.
+    def _list(self, storage_path: str) -> Generator[str, None, None]:
+        """Iterates over the remote directory storage_path and yields any file that is new or
+        has an updated timestamp from when it was last fetched.
 
         Arguments:
+            storage_path (str): Path at a remote location to iterate over
             new_file_callback (Callable, optional): Callback function that
                 is fired each time a new file is fetched
+        """
+        pass
 
-        Returns: count of new files fetched.
+    @abc.abstractmethod
+    def _fetch(self, filepath: str, new_file_callback: Callable) -> None:
+        """Performs actual file fetch from the remote filepath to the internal local_dir
 
+        Arguments:
+            filepath (str): Filepath as a string of the file's remote location
+            new_file_callback (Callable, optional): Callback function that
+                is fired each time a new file is fetched
         """
         pass
+
+    def list_all_generator(self) -> Generator[str, None, None]:
+        """Iterates over all files that need to be fetched"""
+        for storage_path in self.storage_paths:
+            for filepath in self._list(storage_path):
+                yield filepath
+
+    def fetch_all_serial(self, new_file_callback: Callable = lambda: None) -> int:
```

Comment:
is this method just being used in tests? or do you anticipate some future codepaths to use it? if it's the former, should we just put it in test utils instead of making it public to this class?

Reply:
Correct. It was the old `fetch_new` function and isn't used any more. I suppose that might make sense to put it into test utils since its the only place we use it